### PR TITLE
ci: add golang 1.16 to the CI

### PR DESCRIPTION
**Pull request by devimc**

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
lgtm, thanks @devimc!

File: .ci/jenkins_job_build.sh [hunk 1]
```diff
@@ -123,6 +123,9 @@ mkdir -p "${GOPATH}"
 export GOROOT="/usr/local/go"
 export PATH=${GOPATH}/bin:/usr/local/go/bin:/usr/sbin:/sbin:${PATH}
 
+# Support Golang 1.16.x
+export GO111MODULE="auto"
```
Comment (amshinde) [inline .ci/jenkins_job_build.sh:127]: @devimc Can you add in some explanation why this change is required. I see that GO111MODULE is set to on with go 1.16. Would be good to reflect the reasoning behind this in the commit message.

Comment (devimc) [reply .ci/jenkins_job_build.sh:131]: fixed, thanks

**Review by jodh-intel — APPROVED** [review]
Thanks @devimc.

lgtm

**Comment by fidencio** [comment]
Howdy!

I'd like to ask you to rebase your PR atop of current main, as #3668 has been merged and it has a fix that unblocks the CI.

Please, do the rebase and feel free to `@fidencio` in case you do not have the powers to run `/test`.

Sorry for the inconvenience and happy hacking!
